cirq/ops/raw_types.py

```diff
@@ -552,7 +552,7 @@ def _commutes_(
 
         m12 = protocols.unitary_protocol.unitary(circuit12, default=None)
         m21 = protocols.unitary_protocol.unitary(circuit21, default=None)
-        if m12 is None:
+        if m12 is None or m21 is None:
```

Comment:
Looks like a bug. Consider adding a unit test for it.

Reply:
Since the circuits are constructed from the same operations (but in different orders) they would either both admit unitaries or neither, so it's not technically a bug. In any event, the type checker caught it now.